### PR TITLE
fix open by default app and expose default app

### PR DESCRIPTION
**Pull request by wkloucek**
needs: https://github.com/cs3org/cs3apis/pull/157

needed because of https://github.com/owncloud/web/pull/5970#issuecomment-958745928

After this PR the `/app/list` endpoint will provide a default app (if configured):
``` json
    {
      "mime_type": "application/vnd.oasis.opendocument.text",
      "ext": "odt",
      "app_providers": [
        {
          "name": "Collabora",
          "icon": "https://www.collaboraoffice.com/wp-content/uploads/2019/01/CP-icon.png"
        }
      ],
      "name": "OpenDocument",
      "description": "OpenDocument text document",
      "allow_creation": true,
      "default_application": "Collabora"
    }
```

### CODE REVIEW

**Review by glpatcern — APPROVED** [review]
Looks good, clearly having the default semantically explicit is better than just relying on the sorting of the app providers, which was definitely temporary.

At this point we could actually remove the logic to keep the app providers list ordered.

**Comment by wkloucek** [comment]
@glpatcern I updated the CS3apis, can we get this in?

**Comment by glpatcern** [comment]
(did not merge yet because I'm not code owner - will check with @ishank011 also to update the `CODEOWNERS` rules for those subtrees)

**Review by ishank011 — APPROVED** [review]

**Pull request merged**
